components/nimbus/src/enrollment.rs

```diff
@@ -83,7 +89,7 @@ impl ExperimentEnrollment {
         available_randomization_units: &AvailableRandomizationUnits,
         targeting_attributes: &TargetingAttributes,
         experiment: &Experiment,
-        event_store: Arc<Mutex<EventStore>>,
+        #[cfg(feature = "stateful")] event_store: Arc<Mutex<EventStore>>,
```

Comment:
Should we land https://github.com/mozilla/application-services/pull/5374 before we land this?

Reply:
I decided to preemptively merge #5374 into this PR! I made some additional changes to support this, such as pulling the NimbusTargetingHelper out into its own module.